crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -543,6 +546,40 @@ impl Timeline {
         SendAttachment::new(self, path.into(), mime_type, config)
     }
 
+    /// Redacts an event from the timeline.
+    ///
+    /// If it was a local event, this will *try* to cancel it, if it was not
+    /// being sent already. If the event was a remote event, then it will be
+    /// redacted by sending a redaction request to the server.
+    ///
+    /// Returns whether the redaction did happen. It can only return false for
+    /// local events that are being processed.
+    pub async fn redact(
```

Comment:
Does it make sense to add a similar mechanism for `edit`?

- If it's a local echo:
  - Try to abort the sending:
    - If successful, then send the new content,
    - Otherwise, send the new content.
- Otherwise, send the new content.

Reply:
See other comment related to `is_editable()` that's disabled for local echoes: it's mostly a product decision.